frontend/src/components/elements/Metric/Metric.test.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import { Metric as MetricProto } from "src/autogen/proto"
+import { mount } from "src/lib/test_util"
+import Metric, { MetricProps } from "./Metric"
+
+const getProps = (elementProps: Partial<MetricProto> = {}): MetricProps => ({
+  element: MetricProto.create({
+    color: MetricProto.MetricColor.RED,
+    direction: MetricProto.MetricDirection.UP,
+    ...elementProps,
+  }),
+})
+
+describe("Metric element", () => {
+  it("renders metric as expected", () => {
+    const props = getProps()
+    const wrapper = mount(<Metric {...props} />)
+    expect(wrapper).toBeDefined()
+  })
+
+  it("renders correct direction based on props", () => {
+    const props = getProps()
+    const wrapper = mount(<Metric {...props} />)
+    expect(wrapper.text().includes("▲")).toBeTruthy()
+  })
+
+  it("renders correct direction based on props", () => {
+    const props = getProps({
+      color: MetricProto.MetricColor.RED,
```

Comment:
nit: would be good to add a test that checks the CSS of the rendered Metric element sets things to the right color

Same comment for the test below